testsuite/tests/win-unicode/mltest.ml

```diff
@@ -144,9 +144,18 @@ let test_open_in () =
 ;;
 
 let test_getenv () =
+  let equiv l r =
+    assert (l = r);
+    l, r
+  in
   let doit key s =
     Unix.putenv key s;
-    Sys.getenv key, getenvironmentenv key
+    let l = equiv (Sys.getenv key) (getenvironmentenv key) in
+    let r =
+      Unix.putenv key (s ^ s);
+      equiv (Sys.getenv key) (getenvironmentenv key)
+    in
+      l, r
```

Comment:
I don't understand the reason for this change.

Reply:
This test sets a series of code-tortuously-named variables, explicitly testing that the Unicode version of `putenv` is actually working. If those environment variables have been pre-set with the expected values but `putenv` was broken, then the test would pass falsely.

Rather than having to be sure that UTF-8 names were working in `unset` in the `TEST` block, I altered the test itself so that it calls `putenv` twice, so regardless of the initial environment, a failing `putenv` should be detected, since the initial environment cannot yield two different values for the same variable (even on Windows!)